core/services/relay/delegate/delegate.go

```diff
@@ -0,0 +1,129 @@
+package delegate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/relay"
+	"github.com/smartcontractkit/chainlink/core/services/relay/ethereum"
+	"github.com/smartcontractkit/chainlink/core/services/relay/solana"
+	"go.uber.org/multierr"
+)
+
+type delegate struct {
+	relayers relay.Relayers
+}
+
+func NewRelayer(config relay.Config) *delegate {
+	return &delegate{
+		relayers: relay.Relayers{
+			relay.Ethereum: ethereum.NewRelayer(config),
+			relay.Solana:   solana.NewRelayer(config),
+		},
+	}
+}
+
+// A delegate relayer on start will start all relayers it manages.
+func (d delegate) Start() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Start())
+	}
+	return err
+}
+
+// A delegate relayer on close will close all relayers it manages.
+func (d delegate) Close() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Close())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are ready.
+func (d delegate) Ready() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Ready())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are healthy.
+func (d delegate) Healthy() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Healthy())
+	}
+	return err
+}
+
+func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	spec, ok := s.(*job.OffchainReporting2OracleSpec)
+	if !ok {
+		return nil, errors.New("unsuccessful cast to 'job.OffchainReporting2OracleSpec'")
+	}
+
+	choice := relay.Network(spec.Relay)
+	switch choice {
+	case relay.Ethereum:
+		// Notice: we don't use the 'spec.RelayConfig' ATM for 'relay.Ethereum'
```

Comment:
Better put a TODO or FIXME here so it doesnt get forgotten

Reply:
this is already done, removing comment